trust_stores_observatory/store_fetcher/java_fetcher.py

```diff
@@ -0,0 +1,115 @@
+from typing import Tuple, List
+from tempfile import NamedTemporaryFile
+import subprocess
+import re
+import tarfile
+from datetime import datetime
+from urllib.parse import urljoin
+from urllib.request import urlopen
+from urllib.request import urlretrieve
+from urllib.request import Request
+
+from bs4 import BeautifulSoup
+from cryptography.hazmat.primitives import hashes
+
+from trust_stores_observatory.certificates_repository import RootCertificatesRepository
+from trust_stores_observatory.store_fetcher.root_records_validator import RootRecordsValidator
+from trust_stores_observatory.store_fetcher.store_fetcher_interface import StoreFetcherInterface
+from trust_stores_observatory.trust_store import TrustStore, PlatformEnum
+
+class JavaTrustStoreFetcher(StoreFetcherInterface):
+  _BASE_URL = "http://www.oracle.com"
+  _DOWNLOADS_INDEX = "/technetwork/java/javase/downloads/index.html"
+
+  def fetch(self, cert_repo: RootCertificatesRepository, should_update_report: bool=True) -> TrustStore:
+    path_to_cacert = '/lib/security/cacerts'
+    try:
+      headers = {}
+      headers['User-Agent'] = "Mozilla/5.0 Gecko/2010 Firefox/5"
+      headers['Accept'] = 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'
+      headers['Accept-Language'] = 'en-us,en;q=0.5'
+      headers['Accept-Encoding'] = 'gzip, deflate'
+      #cookie set when 'Accept License Agreement' is selected
+      headers['Cookie'] = 'oraclelicense=accept-securebackup-cookie'
+    
+      url,version_num = self._get_latest_package_url()
+      
+      req = Request(url, headers=headers)
+
+      download_content = urlopen(req)  
+      with NamedTemporaryFile(mode='wb') as fh:
+        fh.write(download_content.read())
+        with tarfile.open(name=fh.name, mode='r:gz') as tar_file:
+          cacert_file = tar_file.extractfile('jre-'+version_num+path_to_cacert)
+          with NamedTemporaryFile(mode='wb') as fh2:
+            fh2.write(cacert_file.read()) 
+            fh2.flush()
+            result = subprocess.run(['keytool', '-list', '-v', '-keystore', fh2.name], stdout=subprocess.PIPE, input=b'')
```

Comment:
You can pass `-rfc` and get the certificates out in PEM format, which might be easier to throw into a parser. (Instead of the regexes in `_get_root_records`).

Reply:
Oooh, that's cool! You're right, that will make parsing easier.. all that time spent regexing ;_;

Reply:
...I guess there's no *match*

Reply:
A few things here:
* Using subprocess to run a shell command is not ideal (you have to a lot of work to handle errors and to process/parse the results)
* This assumes `keytool` is installed on the host, which may or may not be true (it won't be true on Windows).

Instead, let's use pyjks (https://github.com/kurtbrose/pyjks) ? It is pure python so it will work on all platforms, and it won't require using a shell.